reflex/app.py

```diff
@@ -175,32 +175,33 @@ def __init__(self, *args, **kwargs):
         self.add_cors()
         self.add_default_endpoints()
 
-        # Set up the Socket.IO AsyncServer.
-        self.sio = AsyncServer(
-            async_mode="asgi",
-            cors_allowed_origins="*"
-            if config.cors_allowed_origins == ["*"]
-            else config.cors_allowed_origins,
-            cors_credentials=True,
-            max_http_buffer_size=constants.POLLING_MAX_HTTP_BUFFER_SIZE,
-            ping_interval=constants.Ping.INTERVAL,
-            ping_timeout=constants.Ping.TIMEOUT,
-        )
+        if self.state is not DefaultState:
+            # Set up the Socket.IO AsyncServer.
```

Comment:
We can move everything in this `if` statement into a function `setup_backend` or something to clean this up as this function is getting quite large now